hypervisor/src/hypervisor/linux.rs

```diff
@@ -119,13 +112,26 @@ impl Hypervisor {
 
         let mut frame_start = Instant::now();
 
+        // retain the first frame start as our sytems t0
+        if self.t0.is_none() {
+            self.t0 = Some(frame_start);
+        }
+
         let sys_time = SYSTEM_START_TIME
             .get()
             .ok_or_else(|| anyhow!("SystemTime was not set"))
             .lev_typ(SystemError::Panic, ErrorLevel::ModuleInit)?;
         sys_time.write(&frame_start).lev(ErrorLevel::ModuleInit)?;
         sys_time.seal_read_only().lev(ErrorLevel::ModuleInit)?;
         loop {
+            // if we are not ment to execute any longer, terminate here
+            match self.terminate_after {
+                Some(terminate_after) if frame_start - self.t0.unwrap() > terminate_after => {
+                    quit::with_code(0)
```

Comment:
I would add a `trace!` statement, indicating that the timeout has been and/or use a different exit code than `0`. Otherwise it is impossible to distinguish to exit cause.

Reply:
I'm not sure that I follow. This would be a graceful exit, thus the return code should be 0. Though you are right request a log print, I guess even info makes sense here (somethint akin `Hypervisor ran for {terminate_after}, now shutting down` should be printed). I will add it.

Reply:
I am with @emilengler in that logging should be added.  
I also like the idea with different exit codes we should also think about exit codes to use for the hypervisor in different scenarios.
 - Ctrl+C 
 - Timeout
 - Health Monitor exit

Reply:
I added a info on the voluntarily quit. Regarding the exit codes, I'm convinced that a timeout should yield a 0 exit code. The timeout is rather an expected run-time of the hypervisor, a termination due to the timeout thus does not equal an error, therefore return code 0.

Should a health monitor event actually be able to shut the hypervisor down?